openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -255,11 +260,25 @@ def objfunc(self, dv_dict):
 
             # Get the objective function evaluations
             for name, obj in iteritems(self.get_objectives()):
-                func_dict[name] = obj
+                if nproc > 1:
+                    owner = system._owning_ranks[name]
+                    if iproc == owner:
+                        func_dict[name] = comm.bcast(obj, root=owner)
+                    else:
+                        func_dict[name] = comm.bcast(None, root=owner)
+                else:
+                    func_dict[name] = obj
 
             # Get the constraint evaluations
             for name, con in iteritems(self.get_constraints()):
-                func_dict[name] = con
+                if nproc > 1:
+                    owner = system._owning_ranks[name]
+                    if iproc == owner:
```

Comment:
If we have to do this stuff to each constraint, do we have to do it to the derivatives too?